src/lib/draw-and-pick.js

```diff
@@ -354,47 +354,47 @@ function getPickedColors(gl, {
   // Make sure we clear scissor test and fbo bindings in case of exceptions
   // We are only interested in one pixel, no need to render anything else
   // Note that the callback here is called synchronously.
-  return glContextWithState(gl, {
+  return withParameters(gl, {
     frameBuffer: pickingFBO,
     framebuffer: pickingFBO,
     scissorTest: {x, y, w: width, h: height}
   }, () => {
 
     // Clear the frame buffer
     gl.clear(GL.COLOR_BUFFER_BIT | GL.DEPTH_BUFFER_BIT);
-    // Save current blend settings
-    const oldBlendMode = getBlendMode(gl);
     // Set blend mode for picking
     // always overwrite existing pixel with [r,g,b,layerIndex]
-    gl.enable(gl.BLEND);
-    gl.blendFuncSeparate(gl.ONE, gl.ZERO, gl.CONSTANT_ALPHA, gl.ZERO);
-    gl.blendEquation(gl.FUNC_ADD);
-
-    // Render all pickable layers in picking colors
-    layers.forEach((layer, layerIndex) => {
-      if (!layer.isComposite && layer.props.visible && layer.props.pickable) {
-
-        // Encode layerIndex with alpha
-        gl.blendColor(0, 0, 0, (layerIndex + 1) / 255);
-
-        layer.drawLayer({
-          uniforms: Object.assign(
-            {renderPickingBuffer: 1, pickingEnabled: 1},
-            layer.context.uniforms,
-            getUniformsFromViewport(layer.context.viewport, layer.props),
-            {layerIndex}
-          )
-        });
-      }
+    const settings = {
+      blend: true,
+      blendFunc: [gl.ONE, gl.ZERO, gl.CONSTANT_ALPHA, gl.ZERO],
+      blendEquation: gl.FUNC_ADD
+    };
+
+    withParameters(gl, settings, () => {
+      // Render all pickable layers in picking colors
+      layers.forEach((layer, layerIndex) => {
+        if (!layer.isComposite && layer.props.visible && layer.props.pickable) {
+
+          // Encode layerIndex with alpha
+          const blendColorSettings = {blendColor: [0, 0, 0, (layerIndex + 1) / 255]};
```

Comment:
Add a `settings` parameter to layer.drawLayer and pass this in there, instead of using another nested `withParameters`.

Even just a `setParameters` call would work since we are already inside a withParameters, value will be restored on exit.